### PR TITLE
Minimum Lockup

### PR DESCRIPTION
**Pull request by bryzettler**

### CODE REVIEW

**Review by ThornM9 — APPROVED** [review]

File: programs/voter-stake-registry/src/state/deposit_entry.rs [hunk 1]
```diff
@@ -158,34 +204,59 @@ impl DepositEntry {
...
-                .checked_mul(remaining as u128)
-                .unwrap()
-                .checked_div(lockup_saturation_secs as u128)
-                .unwrap(),
-        )
-        .unwrap())
+        let total_lockup_secs = self.lockup.total_seconds();
+
+        if total_lockup_secs <= min_required_lockup_saturation_secs {
+            let remaining = min(self.lockup.seconds_left(curr_ts), min_required_lockup_saturation_secs);
```
Comment (ChewingGlass) [inline programs/voter-stake-registry/src/state/deposit_entry.rs]: I'm confused:

  1. Is total seconds less than min required lockup even ever allowed?
  2. How could seconds_left ever be greater than `min_required_lockup_saturation_secs` if the if statement above is true?

File: programs/voter-stake-registry/src/state/deposit_entry.rs [hunk 2]
```diff
@@ -364,7 +435,8 @@ impl DepositEntry {
     pub fn resolve_vesting(&mut self, curr_ts: i64) -> Result<()> {
         let vested_amount = self.vested(curr_ts)?;
         require_gte!(
-            self.amount_initially_locked_native,
+            
+          self.amount_initially_locked_native,
```
Comment (ChewingGlass) [inline programs/voter-stake-registry/src/state/deposit_entry.rs]: Weird spacing happened here?

File: programs/voter-stake-registry/src/instructions/configure_voting_mint.rs [hunk 3]
```diff
@@ -31,13 +31,23 @@ pub struct ConfigureVotingMint<'info> {
...
+/// 
+/// With min_lockup_saturation_secs == 0
 /// ```
 /// vote_weight =
 ///     amount * 10^(digit_shift)
 ///            * (baseline_vote_weight_scaled_factor/1e9
 ///               + lockup_duration_factor * max_extra_lockup_vote_weight_scaled_factor/1e9)
 /// ```
+/// 
+/// With min_lockup_saturation_secs > 0
```
Comment (ChewingGlass) [inline programs/voter-stake-registry/src/instructions/configure_voting_mint.rs]: Is this comment accurate anymore? I don't see this math.

File: programs/voter-stake-registry/src/state/deposit_entry.rs (hunk 1 above)
Comment (bryzettler) [reply programs/voter-stake-registry/src/state/deposit_entry.rs]: No that <= should never happen, ill change the comparison to ==

File: programs/voter-stake-registry/src/instructions/configure_voting_mint.rs (hunk 3 above)
Comment (bryzettler) [reply programs/voter-stake-registry/src/instructions/configure_voting_mint.rs]: No, should be `/// With min_lockup_saturation_secs > 0 && lockup.total_seconds == min_lockup_saturation_secs`

File: programs/voter-stake-registry/src/state/deposit_entry.rs (hunk 2 above)
Comment (bryzettler) [reply programs/voter-stake-registry/src/state/deposit_entry.rs]: fixed

File: programs/voter-stake-registry/src/lib.rs [hunk 4]
```diff
@@ -24,7 +24,7 @@ declare_id!("4Q6WW2ouZ6V3iaNm56MTd5n2tnTm4C5fiH8miFHnAFHo");
 /// The flow for voting with this program is as follows:
 ///
 /// - Create a SPL governance realm.
-/// - Create a governance registry account.
+/// - Create a governan2ce registry account.
```
Comment (bryzettler) [inline programs/voter-stake-registry/src/lib.rs]:
```suggestion
/// - Create a governance registry account.
```

**Pull request closed without merging**